packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -51,248 +47,6 @@ QUnit.module('ember-htmlbars: {{#if}} and {{#unless}} helpers', {
   }
 });
 
-QUnit.test('The `if` helper tests for `isTruthy` if available', function() {
-  view = EmberView.create({
-    truthy: EmberObject.create({ isTruthy: true }),
-    falsy: EmberObject.create({ isTruthy: false }),
-
-    template: compile('{{#if view.truthy}}Yep{{/if}}{{#if view.falsy}}Nope{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `if` helper does not error on undefined', function() {
-  view = EmberView.create({
-    undefinedValue: undefined,
-    template: compile('{{#if view.undefinedValue}}Yep{{/if}}{{#if (unbound view.undefinedValue)}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-});
-
-QUnit.test('The `unless` helper does not error on undefined', function() {
-  view = EmberView.create({
-    undefinedValue: undefined,
-    template: compile('{{#unless view.undefinedValue}}YepBound{{/unless}}{{#unless (unbound view.undefinedValue)}}YepUnbound{{/unless}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'YepBoundYepUnbound');
-});
-
-QUnit.test('The `if` helper does not print the contents for an object proxy without content', function() {
-  view = EmberView.create({
-    truthy: ObjectProxy.create({ content: {} }),
-    falsy: ObjectProxy.create({ content: null }),
-
-    template: compile('{{#if view.truthy}}Yep{{/if}}{{#if view.falsy}}Nope{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `if` helper updates if an object proxy gains or loses context', function() {
-  view = EmberView.create({
-    proxy: ObjectProxy.create({ content: null }),
-
-    template: compile('{{#if view.proxy}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('proxy.content', {});
-  });
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.set('proxy.content', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-function testIfArray(array) {
-  view = EmberView.create({
-    array: array,
-
-    template: compile('{{#if view.array}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.get('array').pushObject(1);
-  });
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.get('array').removeObject(1);
-  });
-
-  equal(view.$().text(), '');
-}
-
-QUnit.test('The `if` helper updates if an array is empty or not', function() {
```

Comment:
Replaced by:

* "it considers empty arrays falsy"

(Now testing re-render, mutation and replacement as well)